synapse/lib/remcycle.py

```diff
@@ -0,0 +1,876 @@
+            nyx_obj = Nyx(api_config=val,
+                          namespace_http_config=self.global_request_headers.get(_namespace)
+                          )
+            self._register_api(name=name, obj=nyx_obj)
+        self.namespaces.add(_namespace)
+
+    def _register_api(self, name, obj):
+        """
+        Register a Nyx object and any corresponding ingest definitions to the
+        cortex.
+        """
+        if name in self.apis:
+            raise Exception('Already registered {}'.format(name))
+        self.apis[name] = obj
+        for gest_name, gest in obj.gests.items():
+            action_name = ':'.join([name, gest_name])
+            # Register the action with the attached cortex
+            ingest_func = s_ingest.register_ingest(core=self.core,
+                                                   gest=gest,
+                                                   evtname=action_name,
+                                                   ret_func=True
+                                                   )
+
+            def gest_glue(event):
+                evtname, event_args = event
+                kwargs = event_args.get('kwargs')
+                resp = kwargs.get('resp')
+                data = resp.get('data')
+                self.core.fire(action_name, data=data)
+
+            # Register the action to unpack the async.Boss job results and fire the cortex event
+            self.on(name=name, func=gest_glue)
+
+            # Store things for later reuse (for deregistartion)
+            self._api_ingests[name].append((action_name, ingest_func, gest_glue))
+
+    def deregister_config(self, namespace):
+        """
+        Remove a given namespace, APIs and any corresponding event handlers
+        which have been snapped into the Hypnos object and its cortex via
+        the register_config API .
+
+        :param namespace: Namespace of objects to remove.
+        :return: None
+        :raises: NoSuchName if the namespace requested does not exist.
+        """
+
+        if namespace not in self.namespaces:
+            raise NoSuchName('Namespace is not registered.')
+
+        self.namespaces.remove(namespace)
+        self.docs.pop(namespace, None)
+        self.global_request_headers.pop(namespace, None)
+
+        apis_to_remove = []
+        for api_name in list(self.apis.keys()):
+            ns, name = api_name.split(':', 1)
+            if ns == namespace:
+                apis_to_remove.append(api_name)
+
+        for api_name in apis_to_remove:
+            self._deregister_api(name=api_name)
+
+    def _deregister_api(self, name):
+        if name not in self.apis:
+            raise NoSuchName('API name not registered.')
+
+        self.apis.pop(name, None)
+
+        funclist = self._api_ingests.pop(name, [])
+        for action_name, ingest_func, gest_glue in funclist:
+            self.off(name, gest_glue)
+            self.core.off(action_name, ingest_func)
+
+    def get_api(self, name):
```

Comment:
getApiNyx()  ( low-first camel case is our default, especially for "external" methods.  Also, since we may want a way to access an API def separately than the instance ( they can be returned over telepath ) we probably want to be a hair more specific about the name ( getApiNyx() leaves room for getApiConf() )